docs/source/tutorials/extending-with-plugins/reader-quickguide.rst

```diff
@@ -0,0 +1,221 @@
+Guide to Writing a GeoIPS Reader
+================================
+
+Overview
+--------
+A GeoIPS reader is a module-based plugin that reads data from specific sources (satellite data, environmental data, etc.) and formats it for use within the GeoIPS framework.
+
+Reader Structure
+---------------
+1. **Call Function** (Required)
+   
+   - Main driver that processes input files
+   - Accepts file names and control parameters
+   - Returns dictionary of xarray Datasets
+
+2. **Read Functions** (Recommended)
+   
+   - Handle actual data reading from files
+   - Convert data into xarray format
+
+3. **Utility Functions** (Optional)
+   
+   - Help with data conversion/manipulation
+   - Format time, coordinates, etc.
+
+Required Output Components
+-------------------------
+
+Xarray Variables
+~~~~~~~~~~~~~~
+- ``latitude``, ``longitude`` (REQUIRED): 2D arrays matching data variables
+- ``time`` (OPTIONAL): 2D array for temporal information
+- Variables of interest: Should match names used in products
+
+Xarray Standard Attributes
+~~~~~~~~~~~~~~~~~~~~~~~~
+**Required Attributes**
+
+- ``source_name``: Identifier used by products
+- ``platform_name``: Data platform
+- ``data_provider``: Source of the data
+- ``start_datetime``: Time of first observation
+- ``end_datetime``: Time of final observation
+- ``interpolation_radius_of_influence``: Used for pyresample-based interpolation
+
+**Optional Attributes**
+
+- ``source_file_names``: List of files used (use base paths or paths with GeoIPS environment variables)
+- ``source_file_attributes``: Dictionary with file names as keys and their attributes as values
+- ``source_file_datetimes``: List of datetime objects for each source file
+- ``area_definition``: Area definition if dataset is registered to one
+- ``registered_dataset``: True if registered to a specific area_definition, False otherwise
+- ``minimum_coverage``: Minimum coverage threshold for product generation
+- ``sample_distance_km``: Used to produce minimal sized imagery for external viewers
+
+Implementation Tips
+------------------
+1. Handle 1D vs 2D variables consistently
+2. Convert time formats to datetime objects
+3. Only read required channels when possible
+4. Implement ``metadata_only`` option for efficiency
+5. Return a dictionary of xarrays with human-readable keys
+6. Handle missing or invalid data appropriately with masks
+7. Apply proper scaling factors and offsets to raw data
+8. Consider memory usage for large datasets
+
+Example Usage
+-------------
+.. code-block:: bash
+
+   # Command line usage
+   --reader_name my_reader_name
```

Comment:
This is lacking context. It's not clear that this is part of a call to `geoips run ...`

Reply:
Changed.